zkevm-circuits/src/evm_circuit/execution/callop.rs

```diff
@@ -145,7 +145,7 @@ impl<F: Field> ExecutionGadget<F> for CallOpGadget<F> {
             );
         });
 
-        cb.condition(call_gadget.has_value.clone(), |cb| {
+        cb.condition(is_call.expr() * call_gadget.has_value.clone(), |cb| {
```

Comment:
When reviewing this I tested the behavior of a CALLCODE with non-zero value in a static context with geth, and there's no error.  I guess this is because CALLCODE sends value to the caller, which is a no-op, so the state isn't changed.  I'm not sure what's the point of the value parameter in CALLCODE.  This would mean that the only callop that can change the state is CALL with non-zero value; all other combinations can be executed successfully in a static context.  Is this your interpretation as well?

Reply:
"This would mean that the only callop that can change the state is CALL with non-zero value" i also think so.